src/main/scala/io/iohk/ethereum/ledger/RewardTeller.scala

```diff
@@ -0,0 +1,69 @@
+package io.iohk.ethereum.ledger
+
+import io.iohk.ethereum.utils.MonetaryPolicyConfig
+
+/**
+  * Calculates rewards for mining blocks and ommers.
+  * Avoids floating point arithmetic. Because of that the formulas may look a bit unintuitive, but the important
+  * thing here is that we want to defer any division to be a single and final operation
+  */
+class RewardTeller(config: MonetaryPolicyConfig) {
```

Comment:
`BlockRewardCalculator` was my initial thought, but I like to give names that are a bit less obvious and more unique, especially for classes encapsulating important and isolated functionality. I find that it makes them stick in your head. I don't think I ever saw a `*Teller` before (think: bank teller), whereas `*Calculator` is so common. 

But if you don't buy it, I can change it.